src/weights.jl

```diff
@@ -193,9 +195,87 @@ pweights(vs::RealArray) = ProbabilityWeights(vec(vs))
     end
 end
 
+# NOTE: A custom constructor is needed because exponential weights must
+# sum to 1, so we want to enforce that on construction of the type
+@weights ExponentialWeights false
+
+"""
+    ExponentialWeights(vs)
+
+Construct an `ExponentialWeights` vector with weight values `vs`, which must sum to 1.
+
+Exponential weights are a common form of temporal weights which assign exponentially
+greater weight to past observations, which in this case corresponds to the tail end of
+the vector.
+"""
+function ExponentialWeights(vs::V) where {T<:Real, V<:AbstractVector{T}}
+    s = sum(vs)
+    s ≈ one(T) || throw(ArgumentError("weight values do not sum to 1 (got $s)"))
+    ExponentialWeights{T, T, V}(vs, s)
+end
+
+"""
+    eweights(n, λ)
+
+Construct an [`ExponentialWeights`](@ref) vector with length `n`,
+where each element in position ``i`` is set to ``λ (1 - λ)^{1 - i}``.
+The entire set of weights are then normalized to sum to 1.
+
+``λ`` is a smoothing factor or rate parameter such that ``0 < λ \\leq 1``.
+As this value approaches 0, the resulting weights will be almost equal,
+while values closer to 1 will put greater weight on the tail elements of the vector.
+
+# Examples
+
+```julia-repl
+julia> eweights(10, 0.3)
+10-element ExponentialWeights{Float64,Float64,Array{Float64,1}}:
+ 0.012458
+ 0.0177971
+ 0.0254245
+ 0.0363207
+ 0.0518867
+ 0.0741238
+ 0.105891
+ 0.151273
+ 0.216104
+ 0.308721
+```
+"""
+function eweights(n::Integer, λ::Real)
+    n > 0 || throw(ArgumentError("cannot construct exponential weights of length < 1"))
+    0 < λ <= 1 || throw(ArgumentError("smoothing factor must be between 0 and 1"))
+    w0 = map(i -> λ * (1 - λ)^(1 - i), 1:n)
+    s = sum(w0)
+    w0 ./= s
+    ExponentialWeights{typeof(s), eltype(w0), typeof(w0)}(w0, s)
+end
+
+"""
+    eweights(vs)
+
+Construct an [`ExponentialWeights`](@ref) vector using the given array.
+"""
+eweights(v::RealVector) = ExponentialWeights(v)
+eweights(v::RealArray) = ExponentialWeights(vec(v))
+
+"""
+    varcorrection(w::ExponentialWeights, corrected=false)
+
+* `corrected=true`: ``\\frac{1}{1 - \\sum {w^2}}``
+* `corrected=false`: ``1``
+"""
+@inline function varcorrection(w::ExponentialWeights, corrected::Bool=false)
+    if corrected
+        1 / (1 - sum(abs2, w.values))
```

Comment:
It comes up as the correction factor for computing the variance of a weighted mean, but aside from that I don't know.

Reply:
Yes, but only of the analytical-weighted mean. And I still don't see a reason to think exponential weights should be considered as analytical weights.

Reply:
I don't know what other kind of weights they would be if not analytic. If you have an exponentially weighted moving average, aren't you weighting successive aggregations, and thus using analytic weights?

Reply:
I don't think so. Analytic weights follow a very specific definition: they represent the number of observations that have been averaged, when giving them equal weights. This determines the variance of the estimator. With exponential weights, the weight does not represent the number of observations which have been averaged, but the weight you give to a certain observation in the computation of the average.

Anyway we need a proof or a reference. We can't just define a method because it might be correct.

Reply:
Hmmm, it's possible that these are more like importance weights in stata, but I'm unclear from the documentation what kind of bias correction, if any, is applied to those weights.

Reply:
None. "Importance weights" is just the Stata name for "we don't know what these weights mean". So no correction can be applied.